### ISSUE TITLE
Misleading error message

### ISSUE THREAD

**Issue by ncannasse** [HaxeFoundation/haxe#4781]
```haxe
// p/A.hx
package p;
class A extends B.T {
}

// p/B.hx
package p;
class B {
}
```

This will report  `Type not found : B` instead of `Module a.B does not define type T` as it should (and does if there's no package involved).

**Comment by Simn** [comment]
Oh god, that probably leads to **that** part of typer.ml...

**Comment by Simn** [comment]
@nadako If you could drive-by check this on your quest in Mordor, that would be greatly appreciated!

**Comment by Simn** [comment]
Err, never mind, this isn't even about expressions in the first place. Sorry for the noise!

**Comment by Simn** [comment]
Closed by #11832.
